### PR TITLE
Do not include voters that have zero voter weight in the election snapshot

### LINKED ISSUE #13938: `MinimumActivesStake` reported as 0
[PR#12889](https://github.com/paritytech/substrate/pull/12889) has introduced a helper for UI/UXs to query the minimum active stake by storing `MinimumActiveStake` in storage. This value is updated everytime the snapshot is created and the election data provider iterates over the `T::VotersList` (bags-list) to select the potential active nominators for the next era.

It has been reported that `MinimumActiveStake` in Polkadot and Kusama has zero in some eras. In the current implementation, there are a few reasons for the minimum active stake to be zero. However, this should no happen and the `MinimumActiveStake` should never be lower than `MinNominatorBond`.

- [x] Ensure that `MinimumActiveStake` is never below `MinNominatorBond`
- [ ] Confirm the decoding issue in Kusama

More notes: https://hackmd.io/NCe-ZVJ2TOC6Tt78bU15BQ

### PR DESCRIPTION
**Pull request by gpestana**
In the current staking system, there can be cases when the `ledger.active` balance is 0 (i.e. the voter weight is 0) or below the `MinNominatorBond` and those nominators will still be included in the election snapshot. E.g. when nominator has unlocking chunks queued (i.e. `ledger.total` balance > 0) and if the nominate has been executed with an older, lower `MinNominatorBond`.

This PR ensures that:
- Voters with `voter_weight == 0` (i.e. `ledger.active == 0`) are not considered by `fn DataProvider::electing_voters` and thus not included in the snapshot.
- `MinActiveStake` can never be zero (although it may be lower than `MinNominatorBond`) 
- The only case where `MinActiveStake` is zero is if the voters list doesn't have any "valid" nominator (i.e. a voter that is decodable and with active stake > 0).

**Changes to EPM mock**
In addition, this PR removes the automatic self staking in the EPM mock `StakingMock::add_target` implementation to be in sync with the `add_target` in the pallet staking implementation. Not including the voters with stake = 0 in the snapshot surfaced the sync issue between test and prod benchmarking setups.

More detailed info: https://hackmd.io/NCe-ZVJ2TOC6Tt78bU15BQ
Closes https://github.com/paritytech/substrate/issues/13938

New version of previously [opened PR](https://github.com/paritytech/substrate/pull/13946)

### CODE REVIEW

File: frame/staking/src/tests.rs [hunk 1]
```diff
@@ -4519,11 +4519,75 @@ mod election_data_provider {
...
+				assert_ok!(Staking::unbond(RuntimeOrigin::signed(101), 100));
+				start_active_era(20);
+
+				// corrupt ledger state by lowering max unlocking chunks bounds.
+				MaxUnlockingChunks::set(1);
+
+				let voters = <Staking as ElectionDataProvider>::electing_voters(None).unwrap();
+				// number of returned voters decreases since ledger entry of stash 101 is now
+				// corrupt.
+				assert_eq!(voters.len(), 4);
```
Comment (Ank4n) [inline frame/staking/src/tests.rs:4581]: 👍

**Review by Ank4n — APPROVED** [review]

**Review by kianenigma — APPROVED** [review]
Vamos 👍

**Comment by kianenigma** [comment]
bot rebase

**Comment by gpestana** [comment]
bot merge

**Pull request merged**